GliaWidgets/ViewController/EngagementViewController.swift

```diff
@@ -32,19 +32,12 @@ class EngagementViewController: ViewController, AlertPresenter {
         viewModel.event(.viewDidDisappear)
     }
 
-    func showBackButton(with style: HeaderButtonStyle, in header: Header) {
-        let back = HeaderButton(with: style, tap: { self.viewModel.event(.backTapped) })
-        back.contentHorizontalAlignment = .left
-        header.setLeftItem(back, animated: false)
-    }
-
-    func showCloseButton(with style: HeaderButtonStyle, in header: Header) {
-        let back = HeaderButton(with: style, tap: { self.viewModel.event(.closeTapped) })
-        back.contentHorizontalAlignment = .right
-        header.setRightItem(back, animated: false)
-    }
-
     private func bind(viewModel: EngagementViewModel, to view: EngagementView) {
+        view.header.endButton.tap = { self.viewModel.event(.closeTapped) }
+        view.header.endScreenShareButton.tap = { self.viewModel.event(.endScreenSharingTapped) }
+        view.header.backButton.tap = { self.viewModel.event(.backTapped) }
+        view.header.closeButton.tap = { self.viewModel.event(.closeTapped) }
```

Comment:
This also looks like a possible place for "retain cycle" and memory leak. Since the strong reference of EngagementViewController has been passed to view child. To be in safe and avoid memory leak perhaps we could use `[weak self]` here. What do you think, does it make sense?

Reply:
I'm honestly not sure if this causing a memory leak is possible, but we can indeed insert `weak` just to be safe, I guess.

Reply:
Done ✅